src/commands/format.cr

```diff
@@ -6,12 +6,26 @@ module Mint
       define_help description: "Formats source files"
 
       define_argument pattern,
-        description: "The pattern which determines which files to format",
-        default: "source/**/*.mint"
+        description: "The pattern which determines which files to format"
 
       def run
         execute "Formatting files" do
-          files = Dir.glob(arguments.pattern.to_s)
+          current =
+            MintJson.parse_current
+
+          format_directories =
+            current.source_directories | current.test_directories
+
+          format_directories_patterns =
+            format_directories.map do |dir|
+              File.join(dir, "**/*.mint")
+            end
+
+          if arguments.pattern.to_s.empty?
+            files = Dir.glob(format_directories_patterns)
+          else
+            files = Dir.glob(arguments.pattern.to_s)
+          end
```

Comment:
One pedantic detail - what about this:

```cr
file_pattern = arguments.pattern.to_s.presence || format_directories_patterns
files = Dir.glob(file_pattern)
```

Reply:
I like that. What is `presence` in this context?

Reply:
See [`String#presence`](https://crystal-lang.org/api/0.32.1/String.html#presence:self?-instance-method) and [`Nil#presence`](https://crystal-lang.org/api/0.32.1/Nil.html#presence-instance-method)

Reply:
Thank You, What is the relationship with `?`?

Reply:
I'm sorry, I don't get the question...

Reply:
I'm sorry, I don't get the question...

Reply:
Does it make sense to do just `presence` or `present?` ?

Reply:
`presence` is the name.

Reply:
Got it.

Reply:
Tried it, turned into a crazy error message, ha
https://pastebin.com/fkjgJ2r4

Reply:
I don't want to spend much time on something so insignificant.